app/javascript/portal/portalThemeHelper.js

```diff
@@ -0,0 +1,146 @@
+import Cookies from 'js-cookie';
+import { adjustColorForContrast } from '../shared/helpers/colorHelper.js';
+
+const SYSTEM_COOKIE = 'system_theme';
+const SELECTED_COOKIE = 'selected_theme';
+
+export const setPortalHoverColor = theme => {
+  if (window.portalConfig.isPlainLayoutEnabled === 'true') return;
+  const portalColor = window.portalConfig.portalColor;
+  const bgColor = theme === 'dark' ? '#151718' : 'white';
+  const hoverColor = adjustColorForContrast(portalColor, bgColor);
+
+  // Set hover color for border and text dynamically
+  document.documentElement.style.setProperty(
+    '--dynamic-hover-color',
+    hoverColor
+  );
+};
```

Comment:
We should be doing this on the backend. For now let's skip it on this PR. We can work on a separate PR with a backend implementation.

I have built a color2k like implementation here: https://gist.github.com/scmmishra/8e2fa38c75bf6fb598ec022018cd271f

We can use this

Reply:
Sure, I will update it in a new PR.

Reply:
https://github.com/chatwoot/chatwoot/pull/8327